### PR TITLE
[link-quality] optimize RSS calculation

### PR DESCRIPTION
**Pull request by simonlingoogle**

### CODE REVIEW

**Review by Irving-cl — APPROVED** [review]
LGTM👍

**Review by abtink — APPROVED** [review]
LGTM. 👍

File: src/core/thread/link_quality.cpp [hunk 1]
```diff
@@ -83,28 +82,10 @@ otError RssAverager::Add(int8_t aRss)
...
-    {
-        // Maintain exponentially weighted moving average using coefficient of (1/2^kCoeffBitShift).
-        // newAverage = + newValue * 1/2^j + oldAverage * (1 - 1/2^j), for j = kCoeffBitShift.
-
-        mAverage = static_cast<uint16_t>(((oldAverage << kCoeffBitShift) - oldAverage + newValue) >> kCoeffBitShift);
-    }
+    mCount += (mCount < (1 << kCoeffBitShift));
+    // Maintain arithmetic mean.
+    // newAverage = newValue * (1/mCount) + oldAverage * ((mCount -1)/mCount)
+    mAverage = static_cast<uint16_t>(((mAverage * (mCount - 1)) + newValue) / mCount);
```
[the remark this answers is not in the document]
Comment (jwhui) [reply src/core/thread/link_quality.cpp:88]: @abtink , just to be clear, I think you meant "Again it does **not** matter..." above?

[the remark this answers is not in the document]
Comment (abtink) [reply src/core/thread/link_quality.cpp:88]: Thanks @jwhui 😃 ... yes exactly. I meant "it does not matter". I edited the comment and fixed it.

**Review by jwhui — APPROVED** [review]

**Pull request merged**